### ISSUE TITLE
Modifying misc with non-str values

### ISSUE THREAD

**Issue by andidyer** [EmilStenstrom/conllu#74]
Hi,

Many compliments on this library. It's by far my preferred one for working with conllu files. I especially appreciate that it's free of dependencies.

One issue that I've encountered is in adding a key-value pair where the value is a float or int to the `misc` field of a token and then serializing it.
My use case is adding token-level analyses (e.g. dependency length) to the `misc` field of a new conllu object to be serialized. 

For example, I might add a new dict to the field as follows:

```python

new_data = {"DL": 2, "Cosine": 0.45}

token["misc"].update(new_data)
```

Although one can modify the dictionary in the `misc` field, adding new key-value pairs, when serializing the following error occurs in `serialize.py`:

```
            fields = []
            for key, value in field.items():
                    value = "_"
                if value == "":
                    fields.append(key)
                    continue

>               fields.append('='.join((key, value)))
E               TypeError: sequence item 1: expected str instance, int found
```

The culprit is this: `fields.append('='.join((key, value)))`. This works okay as long as both the key and the value are `str` type, but it breaks if the value is, for example, an `int` or `float`.

In my fork, I have changed this using an f-string:

```python
def serialize_field(field: T.Any) -> str:
    if field is None:
        return '_'

    if isinstance(field, dict):
        if field == {}:
            return '_'

        fields = []
        for key, value in field.items():
            if value is None:
                value = "_"
            if value == "":
                fields.append(key)
                continue

            fields.append(f'{key}={value}')

        return '|'.join(fields)

    if isinstance(field, tuple):
        return "".join([serialize_field(item) for item in field])

    if isinstance(field, list):
        if len(field[0]) != 2:
            raise ParseException("Can't serialize '{}', invalid format".format(field))
        return "|".join([serialize_field(value) + ":" + str(key) for key, value in field])

    return "{}".format(field)

```

This appears to solve the issue easily, though of course it then means that string representations of arbitrary datatypes could end up in `misc` too, which may be undesirable. Perhaps checking if the value's datatype is in (str, int, float) would constrain this?

**Comment by EmilStenstrom** [comment]
Hi @andidyer! Sorry for missing your issue. I think this is a great improvement, and I don't think we would need to validate the types either. Would you mind writing up a PR too?

**Comment by andidyer** [comment]
Hi,

Sorry, I've just come back to this. Made a pull request now. :)

**Comment by EmilStenstrom** [comment]
For reference: #77

**Comment by andidyer** [comment]
Hi, sorry I'm a bit late addressing these. Will get on it tomorrow. :)

**Comment by EmilStenstrom** [comment]
Thank you @andidyer! I've released [4.5.3](https://pypi.org/project/conllu/) which includes your patch! Thanks!
